### PR TITLE
ISPN-12496 Support port offset for JGroups ports in the server

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12496

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/Bootstrap.java [hunk 1]
```diff
@@ -100,6 +100,13 @@ protected void handleArgumentCommand(String command, String parameter, Iterator<
             // Fall through
          case "--port-offset":
             properties.setProperty(Server.INFINISPAN_PORT_OFFSET, parameter);
+            int offset = Integer.parseInt(parameter);
+            if (!properties.containsKey(Server.JGROUPS_BIND_PORT)) {
+               properties.setProperty(Server.JGROUPS_BIND_PORT, Integer.toString(Server.DEFAULT_JGROUPS_BIND_PORT + offset));
+            }
+            if (!properties.containsKey(Server.JGROUPS_MCAST_PORT)) {
+               properties.setProperty(Server.JGROUPS_MCAST_PORT, Integer.toString(Server.DEFAULT_JGROUPS_MCAST_PORT + offset));
```
Comment (pruivo) [inline server/runtime/src/main/java/org/infinispan/server/Bootstrap.java:108]: this is going to change the UDP & MPING ports. 
Nodes aren't able to see each other anymore without manually setting this property.

Running these commands in 2 different terminals no longer form a cluster
```
$ ./bin/server.sh 
$ ./bin/server.sh -s server2 -o 100
```

File: server/runtime/src/main/java/org/infinispan/server/Bootstrap.java [hunk 2]
```diff
@@ -100,6 +100,13 @@ protected void handleArgumentCommand(String command, String parameter, Iterator<
             // Fall through
          case "--port-offset":
             properties.setProperty(Server.INFINISPAN_PORT_OFFSET, parameter);
+            int offset = Integer.parseInt(parameter);
+            if (!properties.containsKey(Server.JGROUPS_BIND_PORT)) {
+               properties.setProperty(Server.JGROUPS_BIND_PORT, Integer.toString(Server.DEFAULT_JGROUPS_BIND_PORT + offset));
```
Comment (pruivo) [inline server/runtime/src/main/java/org/infinispan/server/Bootstrap.java:105]: JGroups increments the bind_port if it isn't available. Is this a user request? what's the use case?

**Comment by tristantarrant** [comment]
It should mimic the behaviour of the legacy server which applies the offset to all ports, except multicast ports (I've just checked).

**Comment by tristantarrant** [comment]
Repushed without the MCAST_PORT changes

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request merged**